src/app/shared/search-box/input/suggestions/search-suggestions.component.ts

```diff
@@ -0,0 +1,171 @@
+import {AppState} from '../../../../core/store/app.state';
+import {selectAllCollections} from '../../../../core/store/collections/collections.state';
+import {FulltextQueryItem} from '../../query-item/model/fulltext.query-item';
+import {QueryItem} from '../../query-item/model/query-item';
+import {QueryItemType} from '../../query-item/model/query-item-type';
+import {SuggestionsConverter} from './suggestions.converter';
+
+@Component({
+  selector: 'search-suggestions',
+  templateUrl: './search-suggestions.component.html',
+  styleUrls: ['./search-suggestions.component.scss'],
+  // changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SearchSuggestionsComponent implements OnChanges, OnDestroy, OnInit {
+
+  @Input()
+  public queryItems: QueryItem[] = [];
+
+  @Input()
+  public text: string;
+
+  @Input()
+  public moveSelection$: Observable<number>;
+
+  @Input()
+  public useSelection$: Observable<string>;
+
+  @Output()
+  public useSuggestion = new EventEmitter<QueryItem>();
+
+  public suggestions: QueryItem[] = [];
+  public selectedIndex = -1;
+
+  public suggesting = true;
+
+  private searchTerms$ = new Subject<string>();
+  private suggestionsSubscription: Subscription;
+
+  private moveSelectionSubscription: Subscription;
+  private useSelectionSubscription: Subscription;
+
+  constructor(private searchService: SearchService,
+              private store: Store<AppState>) {
+  }
+
+  public ngOnInit() {
+    this.suggestQueryItems();
+    this.subscribeToMoveSelection();
+    this.subscribeToUseSelection();
+  }
+
+  public ngOnChanges(changes: SimpleChanges) {
+    if (changes.hasOwnProperty('queryItems') && this.queryItems) {
+      this.suggesting = !this.queryItems.find(queryItem => queryItem.type === QueryItemType.View);
+    }
+    if (changes.hasOwnProperty('text')) {
+      if (this.text) {
+        this.searchTerms$.next(this.text);
+      } else {
+        this.updateSuggestions([]);
+      }
+    }
+  }
+
+  public ngOnDestroy() {
+    if (this.moveSelectionSubscription) {
+      this.moveSelectionSubscription.unsubscribe();
+    }
+    if (this.useSelectionSubscription) {
+      this.useSelectionSubscription.unsubscribe();
+    }
+    if (this.suggestionsSubscription) {
+      this.suggestionsSubscription.unsubscribe();
+    }
+  }
+
+  private suggestQueryItems() {
+    this.suggestionsSubscription = this.searchTerms$.pipe(
+      startWith(''),
+      debounceTime(300),
+      switchMap(text => this.retrieveSuggestions(text)),
+      withLatestFrom(this.store.select(selectAllCollections)),
+      switchMap(([suggestions, collections]) => SuggestionsConverter.convertSuggestionsToQueryItems(suggestions, collections)),
+      map(queryItems => this.filterViewQueryItems(queryItems)),
+      map(queryItems => this.addFulltextSuggestion(queryItems)),
+      map(queryItems => this.filterUsedQueryItems(queryItems)),
+      catchError(error => {
+        console.error(error);
```

Comment:
Maybe you forgot to remove this :)

Reply:
Not really. It was there before and I think it can stay there so we are able to see the error at least in the console. Because we should not propagate it to the user. That's why we return an observable with an empty array here.